sip-core/src/main/java/de/ikor/sip/foundation/core/premiumsupport/registration/SIPRegistrationProperties.java

```diff
@@ -0,0 +1,108 @@
+package de.ikor.sip.foundation.core.premiumsupport.registration;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.UUID;
+import javax.annotation.PostConstruct;
+import javax.validation.constraints.NotBlank;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.validator.constraints.Length;
+import org.hibernate.validator.constraints.time.DurationMax;
+import org.hibernate.validator.constraints.time.DurationMin;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.annotation.Validated;
+
+/** A configuration class that provides properties specific to an adapter instance. */
+@Data
+@Slf4j
+@Validated
+@Configuration
+@ConfigurationProperties(prefix = "sip.core.backend-registration")
+public class SIPRegistrationProperties {
+
+  @PostConstruct
+  public void setCompositeParameters() {
+    this.checkOutUrl = this.platformUrl + this.checkOutPath;
+    this.checkInUrl = this.platformUrl + this.checkInPath;
+  }
+
+  /**
+   * This is a unique id that identifies an instance of an adapter. The value is generated and can
+   * not be overridden.
+   */
+  @Setter(AccessLevel.NONE)
+  private final UUID instanceId = UUID.randomUUID();
```

Comment:
This unique id looks like something that should be define in a broader scope and probably used on more than one place (actuator endpoints for example). It's kind of buried inside SIPRegistrationProperties. 

That being said, maybe there is a something Spring native that can give us the same result (like spring.application.instance_id in Eureka) or a nicer ID than UUID that can be more descriptive on the UI.

Reply:
We don't have a need for it right now. Instance ID is used only when there is a backend so I'll resolve this for now.